server/src/main/java/io/crate/types/FloatVectorType.java

```diff
@@ -166,6 +166,9 @@ public long valueBytes(float @Nullable [] value) {
 
     @Override
     public TypeSignature getTypeSignature() {
+        if (dimensions == 1) {
+            return new TypeSignature(NAME);
+        }
```

Comment:
Why is this needed?

Reply:
Function is registered with fixed vector dimension 1. 
It's not enough to include the type into `ALLOW_BASENAME_MATCH`. 

Without this change scalar function call fails with 
` The signature type of the based data type parameters can only be: INTEGER_LITERAL_SIGNATURE but was TYPE_SIGNATURE`

I believe it happens because FloatVectorType.getTypeSignature creates `IntegerLiteralTypeSignature` via `TypeSignature.of(dimensions)` and `SignatureBinder.applyBoundVariables` loses that information [here](https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/metadata/functions/SignatureBinder.java#L180).

I removed this check and instead tried adding 
if (typeSignature instanceof IntegerLiteralTypeSignature) {
            return typeSignature;
}

to `SignatureBinder.applyBoundVariables` and tests passed.

Looks like fixing `applyBoundVariables` would be a better option - and we will be able to remove that check also in all BitType/CharacterType/StringType?

See also https://github.com/crate/crate/pull/13278/commits/abc3f7184080049960395960401cb7e93af02659

Reply:
I merged linked PR and removed this "hack"